flixel/util/FlxArrayUtil.hx

```diff
@@ -82,6 +82,25 @@ class FlxArrayUtil
 		return array;
 	}
 
+	/**
+	 * Swaps two elements of an array, which are located at indices `index1` and `index2`
+	 * @param array The array whose elements need to be swapped
+	 * @param index1 The index of one of the elements to be swapped
+	 * @param index2 The index of the other element to be swapped
+	 * @return The array
+	 */
+	@:generic
+	public static inline function swap<T>(array:Array<T>, index1:Int, index2:Int):Array<T>
+	{
+		if(index1 >= 0 && index1 < array.length && index2 >= 0 && index2 < array.length)
```

Comment:
This shouldn't check whether the indices provided are valid, if people provide invalid indices it should just throw whatever error it normally would, we could add `safeSwap` that *does* check. this way we aren't adding needless overhead unless desired by the dev.